packages/react/src/create-strict-api/__tests__/index.test.tsx

```diff
@@ -355,20 +358,54 @@ describe('createStrictAPI()', () => {
         return <button data-testid={testId} className={xcss} />;
       }
 
-      const styles = cssMap({
+      const stylesInvalidRoot = baseCssMap({
         primary: {
-          // @ts-expect-error -- This is not in the `createStrictAPI` schema—this should be a css variable.
           color: 'red',
-          background: 'var(--ds-surface)',
-          '&:hover': { background: 'var(--ds-surface-hover)' },
+          '&:hover': { color: 'var(--ds-text-hover)', background: 'var(--ds-surface-hover)' },
+        },
+      });
+
+      const stylesInvalid = cssMap({
+        primary: {
+          // @ts-expect-error -- This is not valid in the CompiledStrictSchema
+          color: 'red',
+          '&:hover': { color: 'var(--ds-text-hover)', background: 'var(--ds-surface-hover)' },
+        },
+      });
+
+      const stylesValid = cssMap({
+        primary: {
+          color: 'var(--ds-text)',
+          '&:hover': { color: 'var(--ds-text-hover)', background: 'var(--ds-surface-hover)' },
         },
       });
 
       const { getByTestId } = render(
         <>
-          <Button testId="button-1" xcss={styles.primary} />
           <Button
-            testId="button-2"
+            testId="button-invalid-root"
+            // @ts-expect-error -- This conflicts with the custom API, should be a different bg color
+            xcss={stylesInvalidRoot.primary}
+          />
+          <Button
+            testId="button-invalid-root-cx"
+            // @ts-expect-error -- This conflicts with the custom API, should be a different bg color
+            xcss={cx(stylesInvalidRoot.primary, stylesValid.primary)}
+          />
+          <Button
+            testId="button-invalid-strict"
+            // @ts-expect-error -- TODO: This should conflict, but when `cssMap` conflicts, it gets a different type (this has `ApplySchema`, not the raw object), so this doesn't error?  Weird…
+            xcss={stylesInvalid.primary}
```

Comment:
This one's a bit odd, I can't figure it out, only "problem" in this PR, minus the `xcss={{ … }}` removal.

When `cssMap()` has an error in it (this test), the type is different 😕?  Why does this have `ApplySchema` and not the type??
<img width="499" alt="Screenshot 2024-03-06 at 9 32 15 AM" src="https://github.com/atlassian-labs/compiled/assets/136543114/09b9f88d-a739-4041-80a6-d91a4f668816">

Vs. below, where the `cssMap()` type does not error, but the `xcss` errors instead:
<img width="415" alt="Screenshot 2024-03-06 at 9 32 25 AM" src="https://github.com/atlassian-labs/compiled/assets/136543114/a98151ab-1d7d-4a8a-8e3c-678abb2e7b3a">